### PR TITLE
Don't let deleted workflows be default

### PR DESCRIPTION
**Pull request by srallen**
Fixes a scenario where a project sets a workflow as default and then deletes it. This removes the project workflow default if it is the default. 

https://remove-default-if-deleted.pfe-preview.zooniverse.org/

# Review Checklist

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [x] Did you deploy a staging branch?
- [ ] Did you get any type checking errors from [Babel Typecheck](https://github.com/codemix/babel-plugin-typecheck)

## Optional

- [ ] If it's a new component, is it in ES6? Is it clear of warnings from ESLint?
- [ ] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [ ] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?

### CODE REVIEW

File: app/pages/lab/workflow.cjsx [hunk 1]
```diff
@@ -661,6 +661,10 @@ EditWorkflowPage = React.createClass
     if confirmed
       @setState deletionInProgress: true
 
+      if @props.workflow.id is @props.project.configuration?.default_workflow
+        @props.project.update 'configuration.default_workflow': null
+        @props.project.save()
```
Comment (eatyourgreens) [inline app/pages/lab/workflow.cjsx]: Would `delete @props.project.configuration.default_workflow` be better than setting the value to null?

**Comment by eatyourgreens** [comment]
LGTM. I'm not sure if it would be better to delete that configuration key outright, but that isn't a blocking issue.

**Comment by srallen** [comment]
I copied what I did in the checkbox handler: https://github.com/zooniverse/Panoptes-Front-End/blob/master/app/pages/lab/workflow.cjsx#L627-L628

I don't recall if I'm checking for null explicitly anywhere. If it's not needed, I can change both to remove the property entirely.

**Comment by eatyourgreens** [comment]
Deleting the property seems better to me. Off the top of my head, I can't think of a case where we'd explicitly want to check this against `null`, particularly since workflow ID is a string but `null` is an Object.

**Comment by srallen** [comment]
Using the `delete` operator was for some reason causing a 422 on the PUT for the project. Setting the property to undefined seems to work though.

**Comment by eatyourgreens** [comment]
Weird. 😕

**Comment by camallen** [comment]
@srallen ooi what was the payload for the 422 after deleting the key?

**Comment by srallen** [comment]
@camallen the payload for the PUT on the project was `http_cache: true, projects: {}`. My guess is coffeescript was doing something odd with the `delete` operator by making the whole object empty instead of just deleting the one property in the config object.

**Comment by camallen** [comment]
intriguing, one for the memory banks.

**Pull request merged**